java/src/jmri/jmrix/AbstractMRTrafficController.java

```diff
@@ -106,14 +104,14 @@ public AbstractMRTrafficController() {
         mCurrentState = IDLESTATE;
         allowUnexpectedReply = false;
 
-        // We use a shutdown hook here to make sure the connection is left
+
+        // We use a shutdown task here to make sure the connection is left
         // in a clean state prior to exiting.  This is required on systems
         // which have a service mode to ensure we don't leave the system 
-        // in an unusable state (This code predates the ShutdownTask 
-        // mechanisim).  Once the shutdown hook executes, the connection
-        // must be considered closed.
-        shutdownHook = new Thread(new CleanupHook(this));
-        Runtime.getRuntime().addShutdownHook(shutdownHook);
+        // in an unusable state. Once the shutdown task executes, the connection
+        // must be considered permanently closed.
+        
+        InstanceManager.getDefault(ShutDownManager.class).register(shutDownTask = new CleanupTask(this));
```

Comment:
I am going to ask that this not be merged until I have a chance to test this on real hardware.

The whole reason the shutdown hook was put in place to begin with was to prevent JMRI from leaving a connected Lenz command station in service mode if shutdown.  If this doesn't execute correctly, then we leave the Lenz system in an unusable state.  (On Lenz systems, only the device that puts the system into service mode can take it out of service mode).